src/averageAges.js

```diff
@@ -55,7 +79,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter((person) =>
+    onlyWithSon
+      ? person.sex === 'm'
+      && people.some((mother) => {
+        if (person.mother === mother.name) {
+          person['motherBorn'] = mother.born;
+
+          return true;
+        }
+      })
+      : people.some((mother) => {
+        if (person.mother === mother.name) {
+          person['motherBorn'] = mother.born;
+
+          return true;
+        }
+      })
+  );
```

Comment:
super difficult to read. Try to separate some logic into helpers or find better ways to improve its redability